pkg/onchain/sc.go

```diff
@@ -111,10 +111,23 @@ func DeploySC(client *node.Client, wallet wallet.Wallet, contract []byte) (strin
 			return "", fmt.Errorf("waiting SC deployment: %w", err)
 		}
 
+		event := events[0].Data
+		address := strings.Split(event, ":")[1]
+
 		if len(events) > 0 {
-			return strings.Split(events[0].Data, ":")[1], nil
+			//  Catch Run Time Error and return it
+			if strings.Contains(event, "massa_execution_error") {
+				return "", errors.New("runtime error")
+			}
+
+			// Catch Address with generic event parameter, and check if it's a valid address
+			if strings.Contains(event, eventParameter) {
+				if wallet.CheckAddressValidity(address) {
+					return address, nil
+				}
+			}
 		}
 	}
-
-	return "", errors.New("deployment time is out")
+	// If no event received, return an error
+	return "", errors.New("sc deployed successfully but no event received")
```

Comment:
This shouldn't be an error.
Yes there are no events but the contract is successfully deployed.

Reply:
Yes i aggree a lot but i don't know how to change this because we return an Error in the type how can i return a message without an error

How can i do this in Go , return in one case errors and in another case return messages 
![image](https://user-images.githubusercontent.com/10938862/219603001-93e91f76-986d-4b6a-b2a5-eaa4f02106f1.png)

Reply:
You can just return the string but not as an error

Reply:
Yeah after writen this i think about just return the message